sulat/scoring.py

```
@@ -0,0 +1,540 @@
+    try:
+        from urllib.parse import urlparse, urlunparse
+    except ImportError:
+        # Fallback normalization without urllib
+        g_norm = _normalize_url_fallback(str(g).strip())
+        p_norm = _normalize_url_fallback(str(p).strip())
+        return 1.0 if g_norm == p_norm else 0.0
+    
+    try:
+        g_parsed = urlparse(str(g).strip())
+        p_parsed = urlparse(str(p).strip())
+    except Exception:
```

Ruff (0.13.1)

Do not catch blind exception: `Exception`

(BLE001)